### PR TITLE
Add get_by_field method

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Add get_by_field method

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG file.

### CODE REVIEW

File: hpeOneView/resources/settings/appliance_proxy_configuration.py [hunk 1]
```diff
@@ -39,3 +39,24 @@ def __init__(self, connection, data=None):
...
+        """
+        results = self.get_all().data
+        if results:
+            if str(results.get("server", "")).lower() == proxy_ip.lower():
+                new_resource = self.new(self._connection, results)
+            else:
+                new_resource = None
+        return new_resource
+
+    def delete(self):
```
Comment (VenkateshRavula) [inline hpeOneView/resources/settings/appliance_proxy_configuration.py:60]: no need to add this method separately. Existing ```delete``` method in resource class will pick the uri from self.data and will perform delete operation.
Any specific reason to add this method..? Also please add UTs.

Comment (yuvirani) [reply hpeOneView/resources/settings/appliance_proxy_configuration.py:60]: But we dont have uri field in self.data.its None.so here, Im updating self.data['uri'] = self.URI.

Comment (VenkateshRavula) [reply hpeOneView/resources/settings/appliance_proxy_configuration.py:60]: ok. In that case I suggest you not to add ```uri``` field in self.data and try using ```ResourceHelper``` class where you can pass ```uri``` as a param as shown below.
```return self._helper.delete(self.URI, timeout=timeout, force=force)```

File: examples/appliance_proxy_configuration.py [hunk 2]
```diff
@@ -50,10 +50,10 @@
 
 # Get proxy configuration from appliance
 print("\nGet proxy configuration from appliance:\n ")
-proxy_info = proxy.get_all()
```
Comment (AsisBagga) [inline examples/appliance_proxy_configuration.py:53]: we still have get_all() in [library file](https://github.com/HewlettPackard/oneview-python/blob/f1856b68b0b74d3d1b75da2b47e193b601f9fb5a/hpeOneView/resources/settings/appliance_proxy_configuration.py#L39). 
Either remove it from their if not supported or set a separate example.

Comment (yuvirani) [reply examples/appliance_proxy_configuration.py:53]: get_all is being used in get_by_proxy method.. so no need to set separate example.

File: hpeOneView/resources/settings/appliance_proxy_configuration.py (hunk 1 above)
Comment (yuvirani) [reply hpeOneView/resources/settings/appliance_proxy_configuration.py:60]: Done

**Review by VenkateshRavula — APPROVED** [review]

**Review by AsisBagga — APPROVED** [review]

**Pull request merged**